### DISCUSSION TITLE
Have a "Who can use this feature?" call out in our docs

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#27749] [Suggest an Idea]
## Intro

The [GitHub docs, configuring Dependabot version updates](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuring-dependabot-version-updates) have a callout/admonition that explains _who_ can use a feature. For example:

> **Who can use this feature?**
>
> 👤 People with write permissions to a repository can enable or disable Dependabot version updates for the repository.

I like this idea very much, and want to do something similar for our docs.

## Technical details

I'm thinking of creating a custom admonition, like the GitHub one, and using it:

- At the top of stand-alone explanation pages
- In relevant paragraphs
- _Maybe_ for each configuration option, but it's probably better to add a new field "who can use this feature" to the [details table for each config option](https://docs.renovatebot.com/configuration-options/#addlabels)

### Custom admonitions documentation

- [Material for MkDocs, create a _custom_ admonition](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#custom-admonitions).

## Tell me what you think!

What do you maintainers think of this idea? 😄

**Comment by rarkins** [comment]
Can you give an example?

**Reply by HonkingGoose** [reply]
## Screenshot of GitHub docs

![who-can-use-this-feature-on-github-docs](https://github.com/renovatebot/renovate/assets/34918129/da2e4376-7402-4eb6-98c0-b54396396871)

-----

## Sketch for the [Dependency Dashboard explanation page](https://docs.renovatebot.com/key-concepts/dashboard/)

> **Who can use this feature**
>
> 👤  People with read permissions can see the dashboard. People with write permissions can select updates from the Dashboard.

# Introduction

Renovate has a Dependency Dashboard that shows an overview of the state of your repositories' dependencies.

When the Dependency Dashboard is enabled, Renovate will create a new issue in the repository.
This issue has a "dashboard" where you can get an overview of the status of all updates.

Having the Dependency Dashboard also enables the concept of an "approval" workflow for new upgrades, either for selected dependencies (recommended) or even for all.

## Supported platforms

The Dependency Dashboard requires that the host platforms supports the concept of issues with dynamic Markdown checkboxes.

.... rest of the document goes here

**Reply by rarkins** [reply]
This is good, but I'm not sure we should specify *read* privileges as it might require a lot. I think it's useful to document the narrow number of capabilities which require additional privileges (eg adding labels,  checking boxes in PRs, etc)

**Reply by HonkingGoose** [reply]
> This is good, but I'm not sure we should specify read privileges as it might require a lot.

Most of Renovate's features are accessible to _read_ only users, as they can always see the Dependency Dashboard issue, and the PRs that Renovate creates.

So it sounds logical to focus on the things you need elevated permissions for.

Do you want me to make the custom admonition for us then?

**Reply by HonkingGoose** [reply]
@rarkins So are you happy for me to start working on these todos? I did not get a clear go/no go from you yet. :wink: 

- [ ] Creating a new admonition style for **Who can use this feature**
- [ ] Using the admonition for the narrow number of capabilities which need extra privileges (adding labels, checking boxes in PRs, etc)

**Reply by rarkins** [reply]
Can you first work out how many such admonitions would be necessary? I'm undecided if this can be done with regular Notes or if we want a new table row

**Reply by HonkingGoose** [reply]
I'm not sure how many admonitions we need, I would need to go through each page in the docs separately to see what we need. I have a feeling I would want to use the admonition on nearly every page.

I think there's a bigger problem than just creating/using the admonition. I'll explain below.

## Who is the intended reader for a page/section?

The main thing is that most pages do _not_ explain which is the intended reader for that page. Is the page for:

- Beginners?
- Intermediates?
- Advanced users?

Some pages even _mix_ those kinds of information, without marking which sections are for _which_ audience.

## Docs "difficulty" all over the place

The current docs structure leads to a bad difficulty curve. It's very easy to go from a easy page, to deep in the woods technical mumbo jumbo page the next.

Try clicking on the sidebar links in sequence, and note how the difficulty jumps around. Here's the first few pages in sequence:

- Homepage, **easy**
- Reading list, **easy to hard**. But users can choose their difficulty :upside_down_face: 
- Use cases, **medium/hard**. I find this goes in-depth way too soon. It should explain "What can Renovate do for me" first, and only then, for those that want it, explain the technical details.
- Running Renovate, **hard**. Gives a skip link for the Mend Renovate App users, or those who have somebody else hosting Renovate for them.
- Self-hosting examples, **medium**. Users can select which situation applies to them.
- Installing and onboarding, **medium**. I feel we should split this into separate pages:
  - Installing hosted app
  - Installing Renovate when self-hosting
  - Generic page about repository onboarding
  - Generic page about reconfiguration
- Private packages, **hard**. This is really into the weeds about authentication, credentials, encryption.
- Observability, **hard**. This page feels out of place in a "Getting started" section. This feels more of a "deep dive" thing to me.

## Admonitions short-term workaround, real fix is better docs structure

I like the "Who can use this feature" admonition idea, because it helps, in the short term, to explain _who_ is the intended audience for a feature/page/concept. I would want to use it on nearly every page, because:

- It forces us as writers, and maintainers, to be clear about the intended audience of the content.
- It makes us think about splitting difficult pages into separate pages (or sections) for each kind of reader.
- It's the same format each time, so readers will get familiar with this styling and know what to expect from it.
- Readers can self-select if they need the info on the page.
- We can link to different sections on the page for different type of readers.

But the real fix is to re-order and rewrite pages. With the end goal being a smooth ramp up in difficulty, and a clear up-front explanation of each page, and who can use the information on the page/section.

**Reply by rarkins** [reply]
> Some pages even mix those kinds of information, without marking which sections are for which audience.

This is necessary and expected for _reference_ pages, such as config options. I agree with you that other pages which are intended as "guides" should ideally have a target audience in mind
